src/components/NewCommentForm.tsx

```diff
@@ -1,99 +1,161 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import { FORM_FIELDS } from '../utils/constants';
+import { FormFieldsNames } from '../types/FormField';
+import { postComment } from '../api/client';
+import { Post } from '../types/Post';
+import { formatComments } from '../utils/formatData';
+import { Comment } from '../types/Comment';
+import { CommentResponse } from '../types/CommentResponse';
 
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
+type Props = {
+  selectedPost: Post,
+  setNewComment: (comment: Comment) => void,
+  setIsCommentsError: (value: boolean) => void,
+};
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setNewComment,
+  setIsCommentsError,
+}) => {
+  const initialState = {
+    formFields: {
+      name: '',
+      email: '',
+      body: '',
+    },
+    formErrors: {
+      name: false,
+      email: false,
+      body: false,
+    },
+  };
 
-          <span className="icon is-small is-left">
-            <i className="fas fa-user" />
-          </span>
+  const [formFields, setFormFields] = useState(initialState.formFields);
+  const [formErrors, setFormErrors] = useState(initialState.formErrors);
+  const [isLoading, setIsLoading] = useState(false);
 
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
-        </div>
+  const showError = useCallback((key: FormFieldsNames) => {
+    formErrors[key] = true;
+    setFormErrors(prev => ({ ...prev, ...formErrors }));
+  }, [formFields]);
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
-      </div>
+  const validateForm = useCallback(() => {
+    for (const [key, value] of Object.entries(formFields)) {
```

Comment:
<img width="1115" alt="Screenshot 2023-05-16 at 19 09 29" src="https://github.com/mate-academy/react_dynamic-list-of-posts/assets/53228259/768c533e-681a-4ad1-ab59-0d36db86ba84">
you have linter error here